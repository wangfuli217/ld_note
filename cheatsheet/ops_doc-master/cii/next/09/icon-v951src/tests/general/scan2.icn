#SRC: JCON

# test maintenance of scanning environments.

procedure main()
	write("simple")
	"fghjkl" ? {
		write(tab(3))
		write(&pos)
		write(tab(5))
	}
	write("nested")
	"123456" ? {
		write(tab(3))
		"abcdef" ? {
			write(tab(3))
			write(&pos)
			write(tab(5))
		}
		write(&pos)
		write(tab(5))
	}

	write("break")
	every write("98765" ? repeat break upto(&digits))

	write("nested break")
	every write("12345" ? repeat { "67890" ? { write(move(1)); break upto(&digits)}})
	
	write("next")
	every write("98765" ? { every 1 do { "mnbv" ? next }; write(move(2))})

	write("non-local")
	"qwerty" ? {
		every write(foo()) do write(move(1))
	}

	"abcdef" ? {
	    write(&pos)
	    every ="abc" do {
	        write(&pos)
	    }
	    write(&pos)
	}
end

procedure foo()
	suspend move(1)
	"zxc" ? suspend upto(&lcase)
	suspend upto(&lcase)
end
