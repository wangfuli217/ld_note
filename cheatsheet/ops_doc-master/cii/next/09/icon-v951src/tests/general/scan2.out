simple
fg
3
hj
nested
12
ab
3
cd
3
34
break
1
2
3
4
5
nested break
6
1
2
3
4
5
next
98
98
non-local
q
w
1
q
2
w
3
e
4
r
5
t
6
y
1
4
1
